ast/ast.go

```diff
@@ -137,22 +145,37 @@ type ForSpec struct {
 // Apply represents a function call
 type Apply struct {
 	NodeBase
-	Target        Node
-	Arguments     Arguments
-	TrailingComma bool
-	TailStrict    bool
+	Target     Node
+	FodderLeft Fodder
+	Arguments  Arguments
+	// Always false if there were no arguments.
+	TrailingComma    bool
+	TailStrict       bool
+	FodderRight      Fodder
+	TailStrictFodder Fodder
 }
 
 // NamedArgument represents a named argument to function call x=1.
 type NamedArgument struct {
-	Name Identifier
-	Arg  Node
+	NameFodder  Fodder
+	Name        Identifier
+	EqFodder    Fodder
+	Arg         Node
+	CommaFodder Fodder
+}
+
+// CommaSeparatedExpr represents an expression that is an element of a
+// comma-separated list of expressions (e.g. in an array or the arguments of a
+// call)
+type CommaSeparatedExpr struct {
+	Expr        Node
+	CommaFodder Fodder
 }
 
 // Arguments represents positional and named arguments to a function call
 // f(x, y, z=1).
 type Arguments struct {
-	Positional Nodes
+	Positional []CommaSeparatedExpr
```

Comment:
I would probably kept a separate slice of commaFodders in Arguments instead of creating CommaSeparatedExpr struct for this purpose. I think ideally Fodder business would be completely transparent to the code that does not need to care about it. 

It's not a big deal, we can leave it like this AFAIC.

Reply:
Then if someone adds a param to the beginning or something, they would get misaligned.  I think it's better this way - at least you can copy them around opaquely without caring about the fodder.